src/features/offer/components/SameArtistPlaylist/hook/useSameArtistPlaylist.tsx

```diff
@@ -0,0 +1,32 @@
+import { useQuery } from 'react-query'
+
+import {
+  FetchOfferByArtist,
+  fetchOffersByArtist,
+  HitOfferWithArtistAndEan,
+} from 'features/offer/components/SameArtistPlaylist/api/fetchOffersByArtist'
+import { useTransformOfferHits } from 'libs/algolia/fetchAlgolia/transformOfferHit'
+import { useNetInfoContext } from 'libs/network/NetInfoWrapper'
+import { QueryKeys } from 'libs/queryKeys'
+
+export const useSameArtistPlaylist = ({ artist, ean }: FetchOfferByArtist) => {
+  const netInfo = useNetInfoContext()
+  const transformHits = useTransformOfferHits()
+
+  const { data, refetch } = useQuery(
+    [QueryKeys.SAME_ARTIST_PLAYLIST],
+    () => {
+      if (!artist || !ean) return []
+      return fetchOffersByArtist({ artist, ean })
+    },
+    { enabled: !!netInfo.isConnected }
+  )
+
+  if (!data) {
+    return { sameArtistPlaylist: [], refetch }
+  }
+
+  const transformedHits = data?.map(transformHits) as HitOfferWithArtistAndEan[]
+
+  return { sameArtistPlaylist: transformedHits, refetch }
```

Comment:
Je pense que tu peux faire plus simple :
```suggestion
const finalData = data?.map(transformHits) || []
return { sameArtistPlaylist: finalData, refetch }
```

Reply:
Je préfère `hits` que `finalData` mais d'accord avec la suggestion

Reply:
D'accord pour changer le nom de la variable mais j'ai besoin de renvoyé hits en tant que `HitOfferWithArtistAndEan[] ` si je renvoie simplement `[]` il sera de type `AlgoliaHit[]`

Reply:
Oui bah tu peux remettre le `as HitOfferWithArtistAndEan[]` et renommer la variable si tu veux. Je parlais de la méthode pas du résultat final :P